### PR TITLE
Update pytorch to 2.2.2

### PR DESCRIPTION
**Pull request by rikenm1**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Update pytorch to 2.2.2
###### Change Log  

- Change: Update pytorch to 2.2.2

###### Does this affect the toolchain?  

**YES/NO**
NO

###### Test Methodology

- Pipeline build id: [542342](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=542342&view=results)

### CODE REVIEW

File: SPECS/pytorch/pytorch.signatures.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
     "Signatures": {
-        "pytorch-2.0.0-submodules.tar.gz": "81fa75ec16c2228fc75ae8e0c7e5e841665c50581995f48994a490c9a2a87af4",
-        "pytorch-2.0.0.tar.gz": "cecc38b6d4256b810336edfc6119d7a57b701fdf1ba43c50001f31e2724fd8e2"
+        "pytorch-2.2.2-submodules.tar.gz": "74d91f9cbba81848a0c07c718810889c46ca2d24a198444d8e3caca13eea9ffc",
```
Comment (mfrw) [inline SPECS/pytorch/pytorch.signatures.json:3]: Test build seems to be broken

Comment (rikenm1) [reply SPECS/pytorch/pytorch.signatures.json:3]: falak, it's not broken. Actually, it has an unresolved dependency due to runtime dependency on sympy which has a runtime dependency on xorg-x11-server, which was failing.  pytorch -> sympy -> xorg-x11-server.
I recently had a [PR to fix this build failure for xorg-x11-server.](https://github.com/microsoft/azurelinux/pull/8657)

Comment (rikenm1) [reply SPECS/pytorch/pytorch.signatures.json:3]: once yesterday's dailybuild for 3.0-dev is done, this same buddybuild should not cause any failure.

File: SPECS/pytorch/pytorch.spec [hunk 2]
```diff
@@ -22,6 +22,7 @@ BuildRequires:  python3-jinja2
 BuildRequires:  python3-numpy
 BuildRequires:  python3-setuptools
 BuildRequires:  python3-typing-extensions
+BuildRequires:  python3-six
 
 %description
 PyTorch is a Python package that provides two high-level features:
```
Comment (anphel31) [inline SPECS/pytorch/pytorch.spec:28]: my suggestion to fix the ARM64 build out-of-memory issues:
```
%ifarch aarch64
export MAX_JOBS=4
%endif
```

**Review by anphel31 — APPROVED** [review]

**Pull request merged**